query/search.js

```diff
@@ -69,13 +70,15 @@ function generateQuery( clean ){
   }
 
   // focus viewport
-  // @todo: change these to the correct request variable names
-  // @todo: calculate the centroid from the viewport box
-  if( clean.focus && clean.focus.viewport ){
-    var vp = clean.focus.viewport;
+  if( check.number(clean['focus.viewport.min_lat']) &&
+      check.number(clean['focus.viewport.max_lat']) &&
+      check.number(clean['focus.viewport.min_lon']) &&
+      check.number(clean['focus.viewport.max_lon']) ) {
+    // calculate the centroid from the viewport box
     vs.set({
-      'focus:point:lat': vp.min_lat + ( vp.max_lat - vp.min_lat ) / 2,
-      'focus:point:lon': vp.min_lon + ( vp.max_lon - vp.min_lon ) / 2
+      'focus:point:lat': clean['focus.viewport.min_lat'] + ( clean['focus.viewport.max_lat'] - clean['focus.viewport.min_lat'] ) / 2,
+      'focus:point:lon': clean['focus.viewport.min_lon'] + ( clean['focus.viewport.max_lon'] - clean['focus.viewport.min_lon'] ) / 2,
+      'focus:scale': calculateDiagonalDistance(clean) + 'km'
```

Comment:
Does focus:scale expect what is essentially the diameter of a circle centered on focus:point:{lat|lon}? If so that's super convenient. If not we just need to divide by two, right? @missinglink can confirm i'm sure